lib/telemetry_metrics.ex

```diff
@@ -1,5 +1,105 @@
 defmodule Telemetry.Metrics do
   @moduledoc """
   Allows to collect and aggregate Telemetry events over time.
+
+  Metric is an entity transforming a stream of Telemetry events into aggregated values called
+  datapoints. A single metric may produce multiple datapoints, e.g. a counter could report a total
+  number of events it captured as well as number of events emitted during one minute long sliding
+  window.
+
+  ## Data model
+
+  Telemetry metrics provide a multi-dimensional data model. This means that a metric produces
+  multiple sets of datapoints, each set being bound to a unique set of tag values. But what does it
+  mean in practice?
+
+  Let's consider the counter metric. Imagine that you need to count a number of database queries,
+  broken down by the kind of query (`insert`, `delete` etc.) and a table the query is made against.
+  In some metric systems or libraries, you would need to define a metric for each kind of operation
+  and each table. With Telemetry, only one metric is required, and tags can be used to further break
+  down the datapoinst. Let's define such metric:
+
+      definition = Telemetry.Metrics.new(:counter, [:db, :query], tags: [:table, :kind])
+
+  In order for metric to actually work you need to register it first, but let's skip this
+  step for now. Once the metric is up and running, we can start emitting the events:
+
+      Telemetry.execute([:db, :query], 1, %{kind: :insert, table: "users"})
+      Telemetry.execute([:db, :query], 1, %{kind: :select, table: "products"})
+      Telemetry.execute([:db, :query], 1, %{kind: :select, table: "users"})
+      Telemetry.execute([:db, :query], 1, %{kind: :select, table: "users"})
+
+  And the values produced by the metric look as follows:
+
+  | `table` | `kind`  | `total`  | `window`  |
+  |---|---|---|---|
```

Comment:
Minor nitpick: you can indent this table correctly and ExDoc should be able to handle it too.